Multisig/Logic/Models/Safe.swift

```diff
@@ -92,6 +92,10 @@ extension Safe {
         return try? context.fetch(fr).first
     }
 
+    static func by(topic: String) -> Safe? {
+        Safe.all.first {$0.walletConnectSessiontopics.contains(topic)}
```

Comment:
I believe this should have precondition to be executed on main thread, otherwise it might throw an error, that's why I made the original implementation with the precondition

Reply:
Why would this throw an error when not executed on the main thread?

Reply:
`private func handle(request: Request)` has the precondition to not execute on the main thread and then it uses `Safe.by()` to get the safe for the rest of the function. And that needs to be retrieved on the main thread?